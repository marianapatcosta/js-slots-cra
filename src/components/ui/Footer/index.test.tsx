import { render, screen } from '@testing-library/react';
import { Footer } from './index';

describe('Footer', () => {
  it('renders without errors and matches snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Footer />);

    expect(firstChild).toMatchSnapshot();
  });

  it('renders all 3 links elements', () => {
    render(<Footer />);
    const links = screen.getAllByRole('link');
    expect(links.length).toBe(3);
  });
});
