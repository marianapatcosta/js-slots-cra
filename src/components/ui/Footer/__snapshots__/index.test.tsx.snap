// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders without errors and matches snapshot 1`] = `
<footer
  class="footer"
>
  <p>
    @2022 
    footer.developedBy
  </p>
  <div
    class="footer__links"
  >
    <a
      aria-label="footer.findMe"
      href="https://github.com/marianapatcosta"
      rel="nofollow noopener noreferrer"
      target="_blank"
      title="footer.findMe"
    >
      <svg
        aria-hidden="true"
        height="50"
        version="1.1"
        viewBox="0 
 0 438.55 438.55"
        width="50"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="m409.13 114.57c-19.608-33.596-46.205-60.194-79.798-79.8-33.598-19.607-70.277-29.408-110.06-29.408-39.781 0-76.472 9.804-110.06 29.408-33.596 19.605-60.192 46.204-79.8 79.8-19.605 33.595-29.408 70.281-29.408 110.06 0 47.78 13.94 90.745 41.827 128.91 27.884 38.164 63.906 64.572 108.06 79.227 5.14 0.954 8.945 0.283 11.419-1.996 2.475-2.282 3.711-5.14 3.711-8.562 0-0.571-0.049-5.708-0.144-15.417-0.098-9.709-0.144-18.179-0.144-25.406l-6.567 1.136c-4.187 0.767-9.469 1.092-15.846 1-6.374-0.089-12.991-0.757-19.842-1.999-6.854-1.231-13.229-4.086-19.13-8.559-5.898-4.473-10.085-10.328-12.56-17.556l-2.855-6.57c-1.903-4.374-4.899-9.233-8.992-14.559-4.093-5.331-8.232-8.945-12.419-10.848l-1.999-1.431c-1.332-0.951-2.568-2.098-3.711-3.429-1.142-1.331-1.997-2.663-2.568-3.997-0.572-1.335-0.098-2.43 1.427-3.289s4.281-1.276 8.28-1.276l5.708 0.853c3.807 0.763 8.516 3.042 14.133 6.851 5.614 3.806 10.229 8.754 13.846 14.842 4.38 7.806 9.657 13.754 15.846 17.847 6.184 4.093 12.419 6.136 18.699 6.136s11.704-0.476 16.274-1.423c4.565-0.952 8.848-2.383 12.847-4.285 1.713-12.758 6.377-22.559 13.988-29.41-10.848-1.14-20.601-2.857-29.264-5.14-8.658-2.286-17.605-5.996-26.835-11.14-9.235-5.137-16.896-11.516-22.985-19.126-6.09-7.614-11.088-17.61-14.987-29.979-3.901-12.374-5.852-26.648-5.852-42.826 0-23.035 7.52-42.637 22.557-58.817-7.044-17.318-6.379-36.732 1.997-58.24 5.52-1.715 13.706-0.428 24.554 3.853 10.85 4.283 18.794 7.952 23.84 10.994 5.046 3.041 9.089 5.618 12.135 7.708 17.705-4.947 35.976-7.421 54.818-7.421s37.117 2.474 54.823 7.421l10.849-6.849c7.419-4.57 16.18-8.758 26.262-12.565 10.088-3.805 17.802-4.853 23.134-3.138 8.562 21.509 9.325 40.922 2.279 58.24 15.036 16.18 22.559 35.787 22.559 58.817 0 16.178-1.958 30.497-5.853 42.966-3.9 12.471-8.941 22.457-15.125 29.979-6.191 7.521-13.901 13.85-23.131 18.986-9.232 5.14-18.182 8.85-26.84 11.136-8.662 2.286-18.415 4.004-29.263 5.146 9.894 8.562 14.842 22.077 14.842 40.539v60.237c0 3.422 1.19 6.279 3.572 8.562 2.379 2.279 6.136 2.95 11.276 1.995 44.163-14.653 80.185-41.062 108.07-79.226 27.88-38.161 41.825-81.126 41.825-128.91-0.01-39.771-9.818-76.454-29.414-110.05z"
          />
        </g>
      </svg>
    </a>
    <a
      aria-label="footer.findMe"
      href="https://www.linkedin.com/in/marianapatcosta/"
      rel="nofollow noopener noreferrer"
      target="_blank"
      title="footer.findMe"
    >
      <svg
        aria-hidden="true"
        fill="none"
        height="50"
        viewBox="0 0 48 48"
        width="50"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="m0 24c0-13.255 10.745-24 24-24s24 10.745 24 24-10.745 24-24 24-24-10.745-24-24zm16.96-4.1222h-5.4389v16.342h5.4389v-16.342zm0.3583-5.0551c-0.0353-1.6023-1.1811-2.8227-3.0418-2.8227-1.8606 0-3.077 1.2204-3.077 2.8227 0 1.5691 1.1805 2.8246 3.0064 2.8246h0.0348c1.8965 0 3.0776-1.2555 3.0776-2.8246zm19.257 12.027c0-5.0194-2.6832-7.3556-6.2623-7.3556-2.8877 0-4.1805 1.5861-4.9024 2.6988v-2.3146h-5.4396c0.0717 1.5334 0 16.342 0 16.342h5.4396v-9.1266c0-0.4884 0.0353-0.9756 0.1791-1.3253 0.3931-0.9757 1.2881-1.9859 2.7907-1.9859 1.9689 0 2.756 1.4985 2.756 3.6945v8.7429h5.4387l2e-4 -9.3699z"
          fill-rule="evenodd"
        />
      </svg>
    </a>
    <a
      aria-label="footer.findMe"
      href="https://mariana-costa.web.app"
      rel="nofollow noopener noreferrer"
      target="_blank"
      title="footer.findMe"
    >
      <svg
        aria-hidden="true"
        height="50"
        version="1.1"
        viewBox="0 0 512 512"
        width="50"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m502.25 271.57c-18.729 17.964-43.587 27.866-70.036 27.866h-137.86v26.988c0 11.387-9.331 20.723-20.733 20.723h-35.233c-11.397 0-20.726-9.334-20.726-20.723v-26.988h-137.87c-26.445 0-51.308-9.902-70.034-27.866-3.558-3.418-6.815-7.045-9.754-10.852v188.13c0 19.121 15.642 34.749 34.749 34.749h442.51c19.105 0 34.744-15.629 34.744-34.749v-188.13c-2.943 3.807-6.187 7.434-9.753 10.852z"
        />
        <path
          d="m477.26 97.896h-442.51c-19.107 0-34.749 15.641-34.749 34.747v88.637c3.359 14.177 10.788 27.213 21.791 37.763 15.468 14.84 36.07 23.02 57.997 23.02h137.87v-26.983c0-11.4 9.329-20.731 20.726-20.731h35.233c11.402 0 20.733 9.329 20.733 20.731v26.982h137.86c21.926 0 42.534-8.179 57.997-23.02 10.993-10.565 18.435-23.59 21.793-37.773v-88.627c0-19.105-15.639-34.746-34.744-34.746z"
        />
        <path
          d="m195.24 63.146c0-0.986 0.269-2.427 1.536-3.686 1.254-1.26 2.693-1.533 3.686-1.533h111.08c0.978 0 2.422 0.274 3.679 1.533 1.265 1.26 1.533 2.701 1.533 3.686v17.375h29.53v-17.375c-3e-3 -8.891-3.392-17.782-10.171-24.568-6.789-6.789-15.685-10.181-24.573-10.181h-111.08c-8.893 0-17.795 3.392-24.579 10.181-6.779 6.784-10.173 15.677-10.173 24.568v17.375h29.53v-17.375z"
        />
        <path
          d="m273.62 329.77c1.825 0 3.348-1.526 3.348-3.349v-71.344c0-1.823-1.523-3.359-3.348-3.359h-35.236c-1.818 0-3.359 1.536-3.359 3.359v71.344c0 1.823 1.541 3.349 3.359 3.349h35.236z"
        />
      </svg>
    </a>
  </div>
</footer>
`;
